core/markdown_generator.py

```
@@ -250,3 +265,67 @@ def _generate_source_code(self, source_code: Optional[str]) -> str:
+            "examples": element.get("examples", [])
+        }
+        self.search_index.append(metadata)
+
+    def _generate_search_index(self) -> str:
+        """Generate the search index section."""
+        try:
+            search_index_json = json.dumps(self.search_index, indent=2)
+            return "\n".join(
+                [
+                    "## Search Index",
+                    f"```json",
```

Ruff (0.8.0)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)